common/scala/src/main/scala/whisk/core/database/DocumentFactory.scala

```diff
@@ -166,24 +166,29 @@ trait DocumentFactory[W] extends MultipleReadersSingleWriterCache[W, DocInfo] {
 
   def attach[Wsuper >: W](
     db: ArtifactStore[Wsuper],
+    entity: W,
     doc: DocInfo,
     attachmentName: String,
     contentType: ContentType,
     bytes: InputStream)(implicit transid: TransactionId, notifier: Option[CacheChangeNotification]): Future[DocInfo] = {
 
     Try {
       require(db != null, "db undefined")
+      require(entity != null, "entity undefined")
       require(doc != null, "doc undefined")
```

Comment:
Should we get rid of all these null checks? They are for sanity, great, but are they really needed @rabbah ? I'm pretty sure we're `null` free in this area.

Reply:
Yeah we can have a separate cleanup that removes these which date back to the gson serdes.